controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -229,8 +224,31 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Resul
 			sccEncryptionMethod := r.storageClassClaim.Spec.EncryptionMethod
 			_, scIsFSType := existing.Parameters["fsName"]
 			scEncryptionMethod, scHasEncryptionMethod := existing.Parameters["encryptionMethod"]
-			if !((sccType == "sharedfilesystem" && scIsFSType && !scHasEncryptionMethod) ||
-				(sccType == "blockpool" && !scIsFSType && sccEncryptionMethod != scEncryptionMethod)) {
+
+			validStorageClassConfig := true
+			if sccType == "sharedfilesystem" {
+				// check if sc doesn't have parameter "fsName"
+				if !scIsFSType {
+					validStorageClassConfig = false
+				}
+
+				// check if encryption is enabled
+				if scHasEncryptionMethod {
+					validStorageClassConfig = false
+				}
+			} else if sccType == "blockpool" {
+				// check if sc have parameter "fsName"
+				if scIsFSType {
+					validStorageClassConfig = false
+				}
+
+				// check if sc encryption and claim encryption doesn't match
+				if sccEncryptionMethod != scEncryptionMethod {
```

Comment:
```suggestion
				if sccEncryptionMethod == scEncryptionMethod {
```
The De-morgan expansion for the former condition would be:
```diff
- !((sccType == "sharedfilesystem" && scIsFSType && !scHasEncryptionMethod) || (sccType == "blockpool" && !scIsFSType && sccEncryptionMethod != scEncryptionMethod))
+ (sccType != "sharedfilesystem" || !scIsFSType || scHasEncryptionMethod) && (sccType != "blockpool" || scIsFSType || sccEncryptionMethod == scEncryptionMethod)
```
i.e., the `!=` here should be `==`, unless we (intentionally) want to keep it that way?

Reply:
Nevermind, for the updated condition:
```
- !((sccType == "sharedfilesystem" && scIsFSType && !scHasEncryptionMethod) || (sccType == "blockpool" && !scIsFSType && sccEncryptionMethod == scEncryptionMethod))
+ (sccType != "sharedfilesystem" || !scIsFSType || scHasEncryptionMethod) && (sccType != "blockpool" || scIsFSType || sccEncryptionMethod != scEncryptionMethod)
```
The current change is correct.

Reply:
@rexagod Can we verify this by actually testing with live clusters
cc @subhamkrai

Reply:
cc @nb-ohad @agarwal-mudit

Reply:
@Rakshith-R can you explain what seems incorrect to you here?
because I think QE is testing 4.11 installation and we had issues with upgrading from 4.10 to 4.11. So, again can you explain more?

@Madhu-1 fyi

Reply:
> @Rakshith-R can you explain what seems incorrect to you here? because I think QE is testing 4.11 installation and we had issues with upgrading from 4.10 to 4.11. So, again can you explain more?

Just to make sure encrypted storageclassclaim worked.

Reply:
I see so you want to make sure the encryption is working,

Reply:
@Rakshith-R while testing claims with encryption I did find some bug but it was not these if check but it was the wrong parameter in sc. created pr here https://github.com/red-hat-storage/ocs-operator/pull/1753

test restult 
```
oc get storageclassclaim ocs-storagecluster-ceph-rbd-kms -oyaml
apiVersion: ocs.openshift.io/v1alpha1
kind: StorageClassClaim
metadata:
***
  creationTimestamp: "2022-07-15T10:07:43Z"
  finalizers:
  - storageclassclaim.ocs.openshift.io
***
spec:
  encryptionMethod: kms
  type: blockpool
status:
  phase: Ready```